notebooks/ch17/classification_sensitivity_specificity.md

```diff
@@ -0,0 +1,486 @@
+
+For comparison, the ROC curves of `ham_only` and `spam_only` are shown below:
+
+
+```python
+# HIDDEN
+
+ham_only_probabilities = np.zeros(len(y_train))
+spam_only_probabilities = np.ones(len(y_train))
+
+false_positive_rate, sensitivity, thresholds = roc_curve(y_train, ham_only_probabilities, pos_label=1)
+
+plt.step(false_positive_rate, sensitivity, color='b', alpha=0.2,
+         where='post')
+plt.fill_between(false_positive_rate, sensitivity, step='post', alpha=0.2,
+                 color='b')
+plt.xlabel('1 - Specificity (false positive rate)')
+plt.ylabel('Sensitivity')
+plt.title('ham_only ROC Curve')
+```
+
+
+
+
+    Text(0.5,1,'ham_only ROC Curve')
+
+
+
+
+![png](classification_sensitivity_specificity_files/classification_sensitivity_specificity_26_1.png)
+
+
+
+```python
+#HIDDEN
+
+false_positive_rate, sensitivity, thresholds = roc_curve(y_train, spam_only_probabilities, pos_label=1)
+
+plt.step(false_positive_rate, sensitivity, color='b', alpha=0.2,
+         where='post')
+plt.fill_between(false_positive_rate, sensitivity, step='post', alpha=0.2,
+                 color='b')
+plt.xlabel('1 - Specificity (false positive rate)')
+plt.ylabel('Sensitivity')
+plt.title('spam_only ROC Curve')
+```
+
+
+
+
+    Text(0.5,1,'spam_only ROC Curve')
+
+
+
+
+![png](classification_sensitivity_specificity_files/classification_sensitivity_specificity_27_1.png)
+
+
+These ROC curves are unusual (UNSURE OF ANALYSIS HERE) which brings us to our final metric.
+
+### AUC
+
+The **Area Under the Curve (AUC)** is the area under the ROC curve and serves as a single number performance summary of the classifier. Using scikit-learn's [AUC function](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html#sklearn.metrics.roc_auc_score) the AUC for `words_list_model` is:
+
+
+```python
+from sklearn.metrics import roc_auc_score
+
+roc_auc_score(y_train, probabilities)
+```
+
+
+
+
+    0.9057984671441136
+
+
+
+AUC is interpreted as the probability that the classifier will assign a higher probability to a randomly chosen positive data instance than a randomly chosen negative data instance. A perfect AUC value of 1 corresponds to a perfect classifier, and a classifier that randomly labels classes has a diagnoal ROC curve with a AUC of .5. The fact that `words_list_model` has an AUC of .906 means that roughly 91% of the time it is more likely to classify a spam email as spam than a ham email as spam.
+
+
+```python
+# NEED HELP EXPLAINING THIS CELL
+
+roc_auc_score(y_train, ham_only_probabilities)
+```
+
+
+
+
+    0.5
+
+
+
+### Summary
+
+To thoroughly evaluate a model, accuracy, sensitivity, specificity, and AUC ought to be considered together, especially for class-imbalanced datasets. Constructing a confusion matrix helps to visualize the ratio of correct predictions to errors. Based on these results, you can improve a model through further EDA-drive feature selection.
```

Comment:
nit: "model, accuracy, sensitivity, specificity, and AUC" is an odd set of commas. We can probably reword this to avoid that:

> To thoroughly evaluate a model, we consider accuracy, sensitivity, specificity, and AUC. This is particularly important for class-imbalanced datasets, in which ...